File: app/src/main/java/com/g00fy2/circularbackgroundtransition/MainActivity.java
package com.g00fy2.circularbackgroundtransition;

import android.annotation.SuppressLint;
import android.graphics.Color;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import java.util.Random;

public class MainActivity extends AppCompatActivity {

  private CircularBackgroundTransition cbt;
  private Random rnd;

  @SuppressLint("ClickableViewAccessibility") @Override protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    setContentView(R.layout.activity_main);
    cbt = findViewById(R.id.circular_background_transition_view);
    cbt.setOnClickListener(v -> cbt.start(getRandomColor()));
  }

  int getRandomColor() {
    if (rnd == null) rnd = new Random();
    return Color.argb(255, rnd.nextInt(256), rnd.nextInt(256), rnd.nextInt(256));
  }
}

File: app/src/main/java/com/g00fy2/circularbackgroundtransitionsample/MainActivity.kt
package com.g00fy2.circularbackgroundtransitionsample

import android.graphics.Color
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.g00fy2.circularbackgroundtransition.CircularBackgroundTransition
import com.g00fy2.circularbackgroundtransitionsample.R.id
import com.g00fy2.circularbackgroundtransitionsample.R.layout
import java.util.Random

class MainActivity : AppCompatActivity() {

  private val random by lazy { Random() }

  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    setContentView(layout.activity_main)
    findViewById<CircularBackgroundTransition>(id.circular_background_transition_view)?.run {
      setBackgroundColor(randomColor())
      setOnClickListener { startTransition(randomColor()) }
    }
  }

  private fun randomColor() = Color.argb(255, random.nextInt(256), random.nextInt(256), random.nextInt(256))
}